Mathlib/SetTheory/Surreal/Basic.lean

```diff
@@ -384,7 +386,88 @@ theorem one_toGame : toGame 1 = 1 :=
 @[simp]
 theorem nat_toGame : ∀ n : ℕ, toGame n = n :=
   map_natCast' _ one_toGame
-#align surreal.nat_to_game Surreal.nat_toGame
+#align surreal.nat_to_game Surreal.nat_toGamer
+
+@[simp] lemma mk_le_mk {x y : PGame.{u}} {hx hy}: mk x hx ≤ mk y hy ↔ x ≤ y := Iff.rfl
+
+@[simp] lemma out_le_out {x y : Surreal.{u}} : x.out ≤ y.out ↔ x ≤ y := by
+  rw [← Subtype.coe_le_coe, ← mk_le_mk, mk_val_out, mk_val_out]
+
+/-- A small family of games is bounded above. -/
+lemma bddAbove_range_of_small {ι : Type*} [Small.{u} ι] (f : ι → Surreal.{u}) :
+    BddAbove (Set.range f) := by
+  let g : Shrink.{u} ι → PGame.{u} := Subtype.val ∘ Quotient.out ∘ f ∘ (equivShrink.{u} ι).symm
+  have hg (i) : (g i).Numeric := Subtype.prop _
+  let x : Surreal.{u} := mk ⟨Σ i, (g i).LeftMoves, PEmpty,
+    fun x ↦ moveLeft _ x.2, PEmpty.elim⟩ $ .mk (by simp) (fun _ ↦ (hg _).moveLeft _) (by simp)
+  refine ⟨x, Set.forall_range_iff.2 fun i ↦ ?_⟩
+  rw [← (equivShrink ι).symm_apply_apply i, ← out_le_out, ← Subtype.coe_le_coe,
+    PGame.le_iff_forall_lf]
+  simpa using fun j ↦ @moveLeft_lf x.out.val ⟨equivShrink ι i, j⟩
```

Comment:
@timotree3, your suggestion of not having a separate definition for `upperBound` is backfiring. Would you mind having a look (and fixing it because I already spent too much time on this 😇)

Reply:
Indeed. We can't examine the witness of `PGame.bddAbove_range_of_small` and see that it's numeric anymore. For now I've just duplicated the proof. I think it would be cleaner to prove `Surreal.bddAbove_range_of_small` through transitivity with `PGame.bddAbove_range_of_small` and a new lemma `Numeric.exists_ge`, but I think that's too big to do in this PR.

(Here is the cleanup I'm picturing eventually)
```lean
@[simp] lemma _root_.SetTheory.Game.mk_le_mk {x y : PGame.{u}} : (⟦x⟧ : Game) ≤ ⟦y⟧ ↔ x ≤ y :=
  Iff.rfl

@[simp] lemma mk_le_mk {x y : PGame.{u}} {hx hy}: mk x hx ≤ mk y hy ↔ x ≤ y := Iff.rfl

@[elab_as_elim]
theorem inductionOn (x : Surreal) {P : Surreal → Prop} (h : ∀ y h, P (mk y h)) : P x := by
  induction' x using Quotient.inductionOn with x
  rcases x with ⟨x, hx⟩
  apply h

lemma _root_.SetTheory.PGame.Numeric.exists_ge (x : PGame) : ∃ y, x ≤ y ∧ y.Numeric := sorry

/-- A small family of surreals is bounded above. -/
lemma bddAbove_range_of_small {ι : Type*} [Small.{u} ι] (f : ι → Surreal.{u}) :
    BddAbove (Set.range f) := by
  let g : ι → Game.{u} := toGame ∘ f
  have ⟨x, hx⟩ := Game.bddAbove_range_of_small g
  induction' x using Quotient.ind with x
  have ⟨y, hxy, hy⟩ := Numeric.exists_ge x
  refine ⟨mk y hy, fun z ⟨i, hfi⟩ ↦ ?_⟩
  induction' z using inductionOn with z hz
  specialize @hx ⟦z⟧ ⟨i, by simp [hfi]⟩
  simp only [Game.mk_le_mk, mk_le_mk] at hx ⊢
  exact hx.trans hxy
```